### PR TITLE
Restrict IPV4 address as CNAME in DNS changes.

### LINKED ISSUE #843: Disallow IPv4 addresses as CNAME record data in batch change interface
We currently allow users to submit IP addresses that are formatted as fully-qualified domain names (FQDNs) in CNAME record data, which is incorrect since mappings from a record name → IP address should be an A record.

Example of bad data:
```
localhost.some.zone  IN  CNAME  127.0.0.1
```

We need to:
- Add an extra validation in our `BatchChangeValidations` to verify that the `cname` data is _not_ simply an IP address with a terminating dot. Note: A regex like `^[0-9]\.[0-9]\.[0-9]+\.[0-9+]$` may be useful.
- Add a new error message indicating that IP addresses are not valid CNAME targets

-----
Good candidates for files to change:
- Adding new validation: `BatchChangeValidations.scala`
- Adding new error message: `DomainValidationErrors.scala`
- Adding unit tests: `BatchChangeValidationsSpec.scala`
- Adding end-to-end tests: `create_batch_change_test.py`

### PR DESCRIPTION
**Pull request by Jay07GIT**
Fixes #843.

Changes in this pull request:
- Restrict IPV4 address as CNAME in batch changes interface.

### CODE REVIEW

File: modules/api/src/test/scala/vinyldns/api/domain/DomainValidationsSpec.scala [hunk 1]
```diff
@@ -78,6 +78,55 @@ class DomainValidationsSpec
     validateHostName("asterisk.domain*.name.") shouldBe invalid
   }
 
+  property("Shortests fqdn name should be valid") {
```
Comment (nspadaccino) [inline modules/api/src/test/scala/vinyldns/api/domain/DomainValidationsSpec.scala:81]: typo, "Shortests" should be "Shortest"

**Review by nspadaccino — COMMENTED** [review]
Works locally for batch changes and DNS Changes through the portal. But I did notice that I was able to make a cname with an ipv4 address as the record data via the Manage Records tab in a zone, so I think it would be good to restrict that as well.

**Comment by Jay07GIT** [comment]
> record

I noticed the same. Thought to work on new PR for zone page record set. 
Thanks for the suggestion.

File: modules/api/src/test/scala/vinyldns/api/domain/DomainValidationsSpec.scala (hunk 1 above)
Comment (Jay07GIT) [reply modules/api/src/test/scala/vinyldns/api/domain/DomainValidationsSpec.scala]: Corrected typo.

**Review by Aravindh-Raju — COMMENTED** [review]
Same as what Nick suggested. Looks good to me other than that!

**Review by nspadaccino — DISMISSED** [review]
Looks good!

**Review by JoshSEdwards — DISMISSED** [review]

**Review by Aravindh-Raju — APPROVED** [review]
Checked it locally and it works as expected. Looks good to me!

File: modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala [hunk 2]
```diff
@@ -52,6 +52,11 @@ final case class InvalidDomainName(param: String) extends DomainValidationError
       "joined by dots, and terminated with a dot."
 }
 
+final case class InvalidIPv4CName(param: String) extends DomainValidationError {
+  def message: String =
+    s"""Invalid Cname: "$param", valid cname should not be IP address"""
```
Comment (Aravindh-Raju) [inline modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala:57]: A small suggestion, can we update the error message to say it's for `CNAME record data`?
Like:
```
Invalid CNAME record data: "$param", valid CNAME record data should not be IP address.
```

**Review by nspadaccino — DISMISSED** [review]
Something I noticed: While the validation works with the POST /zones/batchrecordchanges route, there isn't validation on the POST /zones/<zoneID>/recordsets route, so I was able to use that to create a CNAME with an IPv4 record data. I think we should have the check on that route as well.

File: modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala (hunk 2 above)
Comment (nspadaccino) [reply modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala:57]: I second Aravindh's suggestion

**Review by arpit4ever — APPROVED** [review]
LGTM

File: modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala (hunk 2 above)
Comment (Jay07GIT) [reply modules/core/src/main/scala/vinyldns/core/domain/DomainValidationErrors.scala]: updated the error message

**Comment by nspadaccino** [comment]
> Something I noticed: While the validation works with the POST /zones/batchrecordchanges route, there isn't validation on the POST /zones//recordsets route, so I was able to use that to create a CNAME with an IPv4 record data. I think we should have the check on that route as well.

We can have this in a new PR

**Review by nspadaccino — APPROVED** [review]

**Pull request merged**